src/qt/intro.cpp

```diff
@@ -133,9 +133,9 @@ Intro::Intro(QWidget *parent, uint64_t blockchain_size, uint64_t chain_state_siz
         ui->prune->setChecked(true);
         ui->prune->setEnabled(false);
     }
-    const unsigned int prune_target_gb = PruneMiBtoGB(prune_target_mib);
-    ui->prune->setText(tr("Discard blocks after verification, except most recent %1 GB (prune)").arg(prune_target_gb ? prune_target_gb : DEFAULT_PRUNE_SIZE_GB));
-    UpdatePruneLabels(prune_target_gb);
+    m_prune_target_gb = prune_target_mib ? PruneMiBtoGB(prune_target_mib) : DEFAULT_PRUNE_SIZE_GB;
```

Comment:
In commit "gui: Fix intro dialog when prune is toogled" (4824eeb91e8ee7ed2e46519f56d860635e7c4818)

I can't figure out what this commit is doing. Could use description of what behavior was before and how it is changed after. Just looking at code I see it is moving away from convention of using 0 pruning size to represent no pruning, but I don't actually see how behavior is changing.

EDIT: Just noticed the new QCheckBox::toggled handler below. Maybe pull refactoring changes out of this commit to make each set of changes more obvious? Also s/toogled/toggled/

Reply:
> I can't figure out what this commit is doing. Could use description of what behavior was before and how it is changed after.

The bug and fixed behavior are described in #17035 OP.

Reply:
This PR description has been updated.